services/viva/microservice/test/lambdas/syncWorkflow.test.ts

```diff
@@ -0,0 +1,145 @@
+import { GetWorkflowResult, LambdaRequest, syncWorkflow } from '../../src/lambdas/syncWorkflow';
+import { EncryptionType } from '../../../types/caseItem';
+import { CaseItem } from '../../src/types/caseItem';
+
+const PK = 'USER#199001011234';
+const SK = 'CASE#11111111-2222-3333-4444-555555555555';
+const WORKFLOW_ID = 'flowId123ABC';
+
+const caseKeys = {
+  PK,
+  SK,
+};
+
+const input: LambdaRequest = {
+  detail: {
+    user: {
+      personalNumber: '199801011234',
+    },
+  },
+};
+
+function createWorkflow(): GetWorkflowResult {
+  return {
+    attributes: {
+      workflowid: WORKFLOW_ID,
+      application: {
+        completiondate: null,
+        completiondescription: '',
+        completionduedate: null,
+        completionreceiveddate: null,
+        completions: null,
+        completionsreceived: null,
+        completionsuploaded: null,
+        islocked: null,
+        islockedwithoutcompletionreceived: null,
+        otherperiod: null,
+        periodenddate: null,
+        periodstartdate: null,
+        receiveddate: null,
+        requestingcompletion: null,
+      },
+    },
+  };
+}
+
+function createCase(workflowId: string = WORKFLOW_ID): CaseItem {
+  return {
+    PK: caseKeys.PK,
+    SK: caseKeys.SK,
+    currentFormId: '123ABC',
+    id: 'caseId',
+    persons: [],
+    state: 'SOME_STATE',
+    expirationTime: 123,
+    createdAt: 123,
+    updatedAt: 456,
+    status: {
+      type: 'active:submitted',
+      name: '',
+      description: '',
+    },
+    forms: {
+      abc123: {
+        answers: [],
+        encryption: {
+          type: EncryptionType.Decrypted,
+        },
+        currentPosition: {
+          currentMainStep: 0,
+          currentMainStepIndex: 0,
+          index: 0,
+          level: 0,
+          numberOfMainSteps: 0,
+        },
+      },
+    },
+    provider: 'VIVA',
+    details: {
+      workflowId: workflowId,
+      period: {
+        startDate: 1,
+        endDate: 2,
+      },
+      completions: {
+        requested: [],
+        description: null,
+        receivedDate: null,
+        dueDate: null,
+        attachmentUploaded: [],
+        isCompleted: false,
+        isRandomCheck: false,
+        isAttachmentPending: false,
+        isDueDateExpired: false,
+      },
+    },
+  };
+}
+
+it('Syncs cases', async () => {
```

Comment:
This is related to all tests:

Instead of typing all dependencies and inputs in every test, I suggest that you write helper function like we do in `syncCaseCompletions.test.ts` and `createDependencies` function.

Otherwise, the tests looks good!